provider/resource_service.go

```diff
@@ -539,7 +539,9 @@ func service(ctx context.Context, d *schema.ResourceData, clientConfig models.Cl
 			for _, thresholdKey := range []string{"time_variate_thresholds", "adaptive_thresholds_is_enabled",
 				"adaptive_thresholding_training_window", "aggregate_thresholds", "entity_thresholds",
 				"time_variate_thresholds_specification"} {
-				itsiKpi[thresholdKey] = thresholdTemplateInterface[thresholdKey]
```

Comment:
Worth to notice that it is sufficient to update just KPI Threshold Id, Splunk will sync policies. I updated just threshold Id, and checked with:
https://www.diffchecker.com/8FZ5s1ma/

Reply:
in most cases, we create threshold templates, then maybe update them, but we don't usually delete them and recreate them. so that should be ok

Reply:
I probably misunderstood the comment, but:
> We create threshold templates, then maybe update them, but we don't usually delete them and recreate them
update of the service or template resource doesn't lead to mentioned error.

Bug will be reproduced in case the service becomes unlinked from the template ID. In case the template is updated, Splunk should sync, rather than lose ID.  
So most probably we faced re-creation via the background processes. In this way we will receive a new ID by title via s3 backend, so the resource config is still valid. But when we try to read real-world-object to create a plan, we receive an error.

With this raws I just mean, that I nullify a lot of threshold-related fields in KPI in this case, but it is ok, just not to put them in body, cause it will be synced anyway.

Direct pushing of the null for some of this results to 500-code error